operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,490 @@
+
+* Create racks by creating `Ceph` buckets and map storage nodes to these buckets using 
+    * "`Ceph` osd crush add-bucket <rack_name> rack"
+    * "`Ceph` osd crush move {bucket-name} {bucket-type}={bucket-name}, [...]"
+* Create a CRUSH rule with rack as a failure domain
+* Map this CRUSH rule to replicated Ceph pools
+
+#### 2.2.2 `Ceph` service zoning
+
+Along with `Ceph` data zoning, `Ceph` services also need to be zoned. These services include:
+
+* Ceph monitor
+* Ceph manager
+* `mds` service for a`dmin-tools`
+* `cephfs`
+
+### 3 Distributing critical services across Kubernetes zones
+
+One of the key ways to ensure that CSM critical services will survive the failure of nodes or a single rack is to spread the replicas of these services across multiple zones and racks.
+
+Some of the services in CSM have already established pod affinities to spread the replicas across nodes. Because the nodes which are picked by the Kubernetes scheduler can be on the same rack, it is necessary to include a topology constraint for these services; this helps the Kubernetes scheduler distribute the replicas across zones. This is achieved using the "Pod Topology Spread Constraints" feature and is implemented through a policy in [Kyverno](../kubernetes/Kyverno.md).
+
+#### 3.1 `kyverno` cluster policy 
+
+A new `kyverno` cluster policy with the `name insert-labels-topology-constraints` has been added to achieve the following:
+
+##### 3.1.1 Add a new label as a selector to identify the pods:
+
+```yaml
+key: rrflag
+value: rr-\<name of the deployment\/statefulset\>
+```
+##### 3.1.2 A Pod Topology Spread Constraint which spreads the pods based on the above selector across zones.
+
+Use the below command to view the policy and know the services for which the policy has been enabled:
+
+```bash
+kubectl get clusterpolicy insert-labels-topology-constraints -o yaml
+```
+   
+#### 3.2 How the `kyverno` policy works before and after failure of nodes or rack
+
+During deployment of the rack resiliency service, the `kyverno` policy is applied to the `Kubernetes` cluster followed by the rolling restart of the services for which the policy has been enforced. This helps `Kubernetes` spread the pods across zones.
+
+Refer to <Link to kubernetes zoning section> for details on rolling restart after `Kubernetes` zoning is enabled.
+
+During the failure of a node or rack when service replicas are restarted by `Kubernetes' the policy helps to spread the replicas being restarted to spread across zones.
+
+**Note:** The policy has been enabled to allow running replicas on the same rack if other racks are not available due to a failure. This ensures that the number of replicas during a failure remains the same.
+
+### 4 Monitor failures and ensure critical services are available across zones
+
+The Resiliency Monitoring Service (RMS) is designed to continuously monitor the health and availability of CSM critical services and CSM management nodes.

```

Comment:
RMS also needs to be added to `glossary.md`, with a link to whatever place is the main documentation for it (this section?)